### DISCUSSION TITLE
x-transition flickering in Safari

### DISCUSSION THREAD

**Discussion by magnusskare** [alpinejs/alpine#2827] [7. Bugs (archived)]
Using `x-transition` on elements with `x-show` seems to make the element flicker just before disappearing. It seems to only happen when the element is transitioning out, just before the element transitions to `display: none`.

The flicker shows the element _pre-transition_ for a split second, so if you've transitioned opacity to 0, it goes back to 1 for a split-second before the element is hidden completely. I've only noticed this in Safari on macOS and iOS.

**Safari, visible flicker just before the element hides:**

https://user-images.githubusercontent.com/56335945/162920761-226a7efe-9c5e-4653-9c9f-613d22241014.mp4

**Chrome, no visible flicker:**

https://user-images.githubusercontent.com/56335945/162920887-059665d6-9002-4a8d-ae49-dcf66ab0a7e0.mp4

Firefox runs similar to Chrome, no issues.

**Comment by frank-dixon** [comment]
@magnusskare I had this issue in Safari only and found that if the opacity transition is paired with another non-opacity transition, the flicker stops. This worked for me (note the **translate-y-0**):

```
x-transition:enter="transition-transform transition-opacity ease-out duration-200"
x-transition:enter-start="opacity-0 transform translate-y-0"
x-transition:enter-end="opacity-100 transform"
x-transition:leave="transition ease-in duration-200"
x-transition:leave-end="opacity-0 transform translate-y-0"
```

@calebporzio

**Comment by sammyaxe** [comment]
@frank-undercovertourist above doesn't seem to help me. Any more insight into this?
